app/controllers/api/v1/sessions_controller.rb

```diff
@@ -26,14 +26,19 @@ def create
     private
 
     def permitted_params
-      params.require(:user).permit(:email, :name, :google_user_id)
+      params.require(:user).permit(:token)
     end
 
     def find_or_create_user
-      @user = User.find_or_initialize_by(email: permitted_params[:email])
+      user_data = fetch_google_oauth_data
+      unless user_data
```

Comment:
This can be done in single line,

`render json: { error: I18n.t('login.failure') }, status: 403 and return unless user_data`

Reply:
applied the changes

Reply:
I don't see this applied.